firebase-inappmessaging-display/src/main/java/com/google/firebase/inappmessaging/display/FirebaseInAppMessagingDisplay.java

```diff
@@ -182,19 +184,7 @@ public void clearFiamListener() {
   @Override
   public void onActivityStarted(final Activity activity) {
     super.onActivityStarted(activity);
-    // Register FIAM listener with the headless sdk.
-    headlessInAppMessaging.setMessageDisplayComponent(
-        (iam, cb) -> {
-          // When we are in the middle of showing a message, we ignore other notifications these
-          // messages will be fired when the corresponding events happen the next time.
-          if (inAppMessage != null || headlessInAppMessaging.areMessagesSuppressed()) {
-            Logging.logd("Active FIAM exists. Skipping trigger");
-            return;
-          }
-          inAppMessage = iam;
-          callbacks = cb;
-          showActiveFiam(activity);
-        });
+    bindFiamToActivity(activity);
```

Comment:
why do we need to bind from activity onstart and onresume? can't we just limit it to one place onResume?